ydb/docs/en/core/concepts/federated_query/s3/external_data_source.md

```diff
@@ -61,7 +61,33 @@ For example, the data schema below describes a schema field named `Year` of type
 Year Int32 NOT NULL
 ```
 
-If a data field is marked as required (`NOT NULL`) but this field is missing in the processed file, the processing of such a file will result in an error. If a field is marked as optional (`NULL`), no error will occur in the absence of the field in the processed file, but the field will take the value `NULL`.
+If a data field is marked as required (`NOT NULL`) but this field is missing in the processed file, the processing of such a file will result in an error. If a field is marked as optional (`NULL`), no error will occur in the absence of the field in the processed file, but the field will take the value `NULL`. Keyword `NULL` is optional in this context.
+
+### Schema inference {#inferring}
+
+Schema inference is available for all [data formats](formats.md#formats) except `raw` and `json_as_string`. It can be useful when the schema contains a large number of fields. In order not to enter those fields manually, use the `WITH_INFER` parameter.:
```

Comment:
```suggestion
Schema inference is available for all [data formats](formats.md#formats) except `raw` and `json_as_string`. It can be helpful when the schema contains a large number of fields. To avoid entering those fields manually, use the `WITH_INFER` parameter:
```